### PR TITLE
py3ify table.py and correct some docstrings

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

- Basically remove six, and future-import. 
- Clean up some docstrings on the way.

### CODE REVIEW

File: lib/matplotlib/table.py [hunk 1]
```diff
@@ -461,8 +456,7 @@ def auto_set_column_width(self, col):
         self.stale = True
 
     def _auto_set_column_width(self, col, renderer):
-        """ Automagically set width for column.
-        """
+        """Automatically set width for column."""
```
Comment (anntzer) [inline lib/matplotlib/table.py]: I think automagically is fine here, we're allowed some humor for private docstrings at least...

File: lib/matplotlib/table.py [hunk 2]
```diff
@@ -511,14 +505,13 @@ def set_fontsize(self, size):
...
-        for cell in six.itervalues(self._cells):
+        for cell in self._cells.values():
             cell.set_fontsize(size)
         self.stale = True
 
     def _offset(self, ox, oy):
-        'Move all the artists by ox,oy (axes coords)'
-
-        for c in six.itervalues(self._cells):
+        """Move all the artists by ox,oy (axes coords)."""
```
Comment (anntzer) [inline lib/matplotlib/table.py]: space after comma

File: lib/matplotlib/table.py (hunk 1 above)
Comment (timhoffm) [reply lib/matplotlib/table.py]: Not sure it was intended, though. All other similar methods around this are using "automatically" and 'g' and 't' are adjacent on the keyboard, so I assume this is just a funny typo.

Comment (anntzer) [reply lib/matplotlib/table.py]: I don't really care tbh, changing is fine.

**Pull request merged**